Automattic/woocommerce-payments
# feat: opt-out survey endpoint

Fixes https://github.com/Automattic/woocommerce-payments/issues/2310

#### Changes proposed in this Pull Request









#### Testing instructions





- Use in conjunction with https://github.com/Automattic/woocommerce-payments/issues/2309 (apply the patch to your dev server & change your host file to point `public-api.wordpress.com` to your sandbox's IP address)
- In WCPay Dev tools, ensure the "Enable UPE settings changes" & "Enable grouped settings" checkboxes are checked
- Go to the settings page: http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments
- In the browser console, test with:
- `wp.apiFetch({ path: '/wc/v3/payments/upe_survey', method: 'POST' })` => should make a network request that fails with an error
- `wp.apiFetch({ path: '/wc/v3/payments/upe_survey', method: 'POST', data: { 'why-disable': 'theme-compatibility', comments: 'test' } })` => should make a network request that sends data to the survey tool

-------------------
